3.x: Update guava to 33.3.1-jre

dkropachev opened the pull request:
The goal is to fixe ~8 vulnurabilities that comming from guava dependency.
This PR does the following:
1. Drops `GuavaCompatibility`
2. Updates documentation to reflect that it targets only one version of guava
3. Updates guava to `33.3.1-jre`

dkropachev committed:
Update guava to 33.3.1-jre

It fixes ~8 vulnurabilities that comming from guava dependency.

nikagra commented on driver-core/src/main/java/com/datastax/driver/core/GuavaCompatibility.java:
```diff
@@ -188,84 +179,18 @@ public abstract <I, O> ListenableFuture<O> transformAsync(
-      // Guava < 22.0
-      return hostAndPort.getHostText();
-    }
+    // Guava >= 20.0
```
Javadoc comment states that "driver is compatible with Guava 19.0 or higher", but the comment above contradicts it. Is Guava 19.0 supported?

dkropachev replied:
Whole file is removed.

nikagra commented on driver-core/src/main/java/com/datastax/driver/core/GuavaCompatibility.java:
```diff
@@ -188,84 +179,18 @@ public abstract <I, O> ListenableFuture<O> transformAsync(
    * <p>The method {@code HostAndPort.getHostText} has been replaced with {@code
```
I think reference to removed `HostAndPort.getHostText` can be dropped now.

dkropachev replied:
i have dropped `GuavaCompatibility.java` ancient guava version are in the past, this layer is not needed anymore

dkropachev committed:
Drop GuavaCompatibility

Driver do not support old guavas anymore, on modern guavas API
look-alike, we don't need this layer anymore.

dkropachev commented:
@sylwiaszunejko , could you please take a look at it
